assets/in

```diff
@@ -16,26 +18,49 @@ fi
 
 # parse incoming config data
 payload=$(cat)
-bucket=$(echo "${payload}" | jq -r '.source.bucket')
+
+bucket="$(echo "${payload}" | jq -r '.source.bucket')"
+if [ -z "${bucket}" ]; then
+  echo "No bucket supplied"
+  exit 1
+fi
+
 endpoint_url="$(echo "${payload}" | jq -r '.source.endpoint // ""')"
 region="$(echo "${payload}" | jq -r '.source.region // "us-east-1"')"
 use_v4="$(echo "${payload}" | jq -r '.source.use_v4 // ""')"
-path=$(echo "${payload}" | jq -r '.source.path // ""')
-suffix=$(echo "${payload}" | jq -r '.source.suffix // ""')
+path="$(echo "${payload}" | jq -r '.source.path // ""')"
+if [ "${path}" = '/' ]; then
+  path=""
+fi
+suffix="$(echo "${payload}" | jq -r '.source.suffix // ""')"
 encryption_key="$(echo "${payload}" | jq -r '.source.encryption_key // ""')"
 skip_ssl_verification="$(echo "$payload" | jq -r '.source.skip_ssl_verification // ""')"
 ca_bundle="$(echo "$payload" | jq -r '.source.ca_bundle // ""')"
+minio_identifier="minio" # used to check if endpoint_url contains 'minio'
 
 # export for `aws` cli
 AWS_ACCESS_KEY_ID=$(echo "$payload" | jq -r '.source.access_key_id')
+if [ -z "${AWS_ACCESS_KEY_ID}" ] || [ "${AWS_ACCESS_KEY_ID}" = "null" ]; then
+  if [ "${endpoint_url#*$minio_identifier}" != "${endpoint_url}" ]; then # if no creds are supplied and endpoint is for minio, exit
+    echo "No access_key_id supplied, needed if accessing minio."
+    exit 1
+  fi
+fi
 AWS_SECRET_ACCESS_KEY=$(echo "$payload" | jq -r '.source.secret_access_key')
+if [ -z "${AWS_SECRET_ACCESS_KEY}" ] || [ "${AWS_SECRET_ACCESS_KEY}" = "null" ]; then
+  if [ "${endpoint_url#*$minio_identifier}" != "${endpoint_url}" ]; then
```

Comment:
I wouldnt differentiate beteen minio and aws. It doesn't matter as both are an implemetation of s3 api. We do `aws s3 ls ` below to check if path exists. This is enough to figure out if correct credentials were supplied.